core/store/src/lib.rs

```diff
@@ -291,28 +291,79 @@ impl NodeStorage {
     }
 }
 
+pub fn retrieve_shard_uid_from_db_key(key: &[u8]) -> io::Result<ShardUId> {
+    // TODO(reshardingV3) Consider changing the Error type to `StorageError`?
+    // Would need changing error types for `Store` methods as well.
+    ShardUId::try_from(&key[..8])
+        .map_err(|e| io::Error::other(format!("Could not retrieve ShardUId from db key: {}", e)))
+}
+
+pub fn replace_shard_uid_key_prefix(key: &[u8], shard_uid: ShardUId) -> [u8; 40] {
+    let mut mapped_key = [0u8; 40];
+    mapped_key[..8].copy_from_slice(&shard_uid.to_bytes());
+    mapped_key[8..].copy_from_slice(&key[8..]);
+    mapped_key
+}
+
 impl Store {
     pub fn new(storage: Arc<dyn Database>) -> Self {
         Self { storage }
     }
 
+    /// Underlying `get()` implementation for all columns.
+    fn get_impl(&self, column: DBCol, key: &[u8]) -> io::Result<Option<DBSlice<'_>>> {
```

Comment:
nice

Reply:
maybe debug assert that cold != State and mention that in the comment

Reply:
moved to `TrieStoreAdapter` in a later commit